ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -414,51 +414,65 @@ const Page: NextPage<Props> = ({
                       if (connectionOptions[opt.key]?.type === "typeahead") {
                         return (
                           <>
-                            <Typeahead
-                              id="typeahead"
-                              labelKey="key"
-                              disabled={loading || loadingOptions}
-                              options={connectionOptions[opt.key]?.options.map(
-                                (k, idx) => {
-                                  return {
-                                    key: k,
-                                    descriptions:
-                                      connectionOptions[k]?.descriptions[idx],
-                                  };
-                                }
-                              )}
-                              placeholder={
-                                opt.placeholder || `Select ${opt.key}`
-                              }
-                              renderMenuItemChildren={(opt, props, idx) => {
-                                return [
-                                  <span key={`opt-${idx}-key`}>
-                                    {opt.key}
-                                    <br />
-                                  </span>,
-                                  <small
-                                    key={`opt-${idx}-descriptions`}
-                                    className="text-small"
-                                  >
-                                    {opt.descriptions ? (
-                                      <em>
-                                        Descriptions:{" "}
-                                        {opt.descriptions.join(", ")}
-                                      </em>
-                                    ) : null}
-                                  </small>,
-                                ];
+                            <Controller
+                              control={control}
+                              name={`source.options.${opt.key}`}
+                              render={({ field: { onChange } }) => {
```

Comment:
Because `Typeahead`'s props aren't compatible with the props `react-hook-form`'s `register` function returns, we need to wrap it in a `Controller` with a custom render function